### PR TITLE
UCP: fix leaked UCT EPs after ucp_ep_close_nb(UCP_EP_CLOSE_MODE_FORCE)

### PR DESCRIPTION
**Pull request by evgeny-leksikov**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5598/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@alex-mikheev @brminich pls review

File: test/gtest/ucp/test_ucp_peer_failure.cc [hunk 1]
```diff
@@ -202,6 +210,13 @@ void test_ucp_peer_failure::test_force_close()
     } else {
         EXPECT_FALSE(UCS_PTR_IS_ERR(close_req));
     }
+
+    mem_after = mallinfo();
+    /* Too low chance to predict memory consumption on wire up for all TLS */
+    if (GetParam().variant != FAIL_IMMEDIATELY) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_peer_failure.cc]: seems like it checks if something was release, but not if everything was released?
does this test fail if w/o fix in UCP?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: yes, for udrc TL the consumption was constant

**Comment by evgeny-leksikov** [comment]
@alex-mikheev typically the other side is already dead. otherwise, according to documentation remote side should be ready to handle if something will not be completed. if remote side will post sends, it will be completed with peer timeout error.

**Comment by alex-mikheev** [comment]
Do we already have tests that cover that ?

**Comment by evgeny-leksikov** [comment]
No. We have peer failure tests to cover any "unexpected" remote closure, if you mean this.

**Pull request merged**